ecs/sdk.go

```diff
@@ -317,29 +317,113 @@ func (s sdk) ListStacks(ctx context.Context, name string) ([]compose.Stack, erro
 	}
 	stacks := []compose.Stack{}
 	for _, stack := range cfStacks.Stacks {
+		skip := true
 		for _, t := range stack.Tags {
 			if *t.Key == compose.ProjectTag {
-				status := compose.RUNNING
-				switch aws.StringValue(stack.StackStatus) {
-				case "CREATE_IN_PROGRESS":
-					status = compose.STARTING
-				case "DELETE_IN_PROGRESS":
-					status = compose.REMOVING
-				case "UPDATE_IN_PROGRESS":
-					status = compose.UPDATING
-				}
-				stacks = append(stacks, compose.Stack{
-					ID:     aws.StringValue(stack.StackId),
-					Name:   aws.StringValue(stack.StackName),
-					Status: status,
-				})
+				skip = false
 				break
 			}
 		}
+		if skip {
+			continue
+		}
+		status := compose.RUNNING
+		reason := ""
+		switch aws.StringValue(stack.StackStatus) {
+		case "CREATE_IN_PROGRESS":
+			status = compose.STARTING
+		case "DELETE_IN_PROGRESS":
+			status = compose.REMOVING
+		case "UPDATE_IN_PROGRESS":
+			status = compose.UPDATING
+		}
+		if status == compose.STARTING {
+			if err := s.CheckStackState(ctx, aws.StringValue(stack.StackName)); err != nil {
+				status = compose.FAILED
+				reason = err.Error()
+			}
+		}
+		stacks = append(stacks, compose.Stack{
+			ID:     aws.StringValue(stack.StackId),
+			Name:   aws.StringValue(stack.StackName),
+			Status: status,
+			Reason: reason,
+		})
+
 	}
 	return stacks, nil
 }
 
+func (s sdk) CheckStackState(ctx context.Context, name string) error {
+	resources, err := s.CF.ListStackResourcesWithContext(ctx, &cloudformation.ListStackResourcesInput{
+		StackName: aws.String(name),
+	})
+	if err != nil {
+		return err
+	}
+	services := []*string{}
+	serviceNames := []string{}
+	var cluster *string
+	for _, r := range resources.StackResourceSummaries {
+		if aws.StringValue(r.ResourceType) == "AWS::ECS::Cluster" {
+			cluster = r.PhysicalResourceId
+			continue
+		}
+		if aws.StringValue(r.ResourceType) == "AWS::ECS::Service" {
+			if r.PhysicalResourceId == nil {
+				continue
+			}
+			services = append(services, r.PhysicalResourceId)
+			serviceNames = append(serviceNames, *r.LogicalResourceId)
+		}
+	}
+	for i, service := range services {
+		err := s.CheckTaskState(ctx, aws.StringValue(cluster), aws.StringValue(service))
+		if err != nil {
+			return fmt.Errorf("%s error: %s", serviceNames[i], err.Error())
+		}
+	}
+	return nil
+}
+
+func (s sdk) CheckTaskState(ctx context.Context, cluster string, serviceName string) error {
```

Comment:
AFAICT this will also return tasks from a previous `compose up`, we need to filter tasks by last time service was updated by compose-cli (maybe based on TaskDefinition ID?)